pkg/node/volumemgr.go

```diff
@@ -1213,3 +1215,38 @@ func (m *VolumeManager) addVolumeStatusAnnotation(drive *drivecrd.Drive, volumeN
 	}
 	drive.Annotations[annotationKey] = status
 }
+
+// handleExpandingStatus handles volume CR with Expanding status, it calls ExpandLV to expand volume
+// To get logical volume name it use LVM provisioner function GetVolumePath
+// Receive context, volume CR
+// Return ctrl.Result, error
+func (m *VolumeManager) handleExpandingStatus(ctx context.Context, volume *volumecrd.Volume) (ctrl.Result, error) {
+	ll := m.log.WithFields(logrus.Fields{
+		"method": "handleExpandingStatus",
+	})
+	volumePath, err := m.provisioners[p.LVMBasedVolumeType].GetVolumePath(volume.Spec)
+	if err != nil {
+		ll.Errorf("Failed to get volume path, err: %v", err)
+		return ctrl.Result{Requeue: true}, err
+	}
+	mbSize, err := util.ToSizeUnit(volume.Spec.Size, util.BYTE, util.MBYTE)
+	if err != nil {
+		ll.Errorf("Failed to convert %d to %v, err: %v", volume.Spec.Size, util.MBYTE, err)
+		return ctrl.Result{Requeue: true}, err
+	}
+	if err = m.lvmOps.ExpandLV(volumePath, mbSize); err != nil {
+		volume.Spec.CSIStatus = apiV1.Failed
+		if updateErr := m.k8sClient.UpdateCRWithAttempts(ctx, volume, 10); updateErr != nil {
+			ll.Errorf("Unable to change volume %s status to %s, error: %v.",
+				volume.Name, volume.Spec.CSIStatus, updateErr)
+			return ctrl.Result{Requeue: true}, updateErr
+		}
+		return ctrl.Result{Requeue: true}, err
+	}
+	volume.Spec.CSIStatus = apiV1.Expanded
```

Comment:
Could you please clarify why we need `Expanded` status? Do we need to know that volume resized after the expansion operation was fully completed? Can we use `created` as a generic status, which means "volume exists but not published"?

Reply:
`ControllerExpandVolume` handles both online and offline volume expansion, so volume can be in published, created or volume_ready state. As I see we can't just set `created`, because the previous state of the volume could be published for example. I assume, we need to know if the volume was successfully expanded, so we can set its previous state after expansion.